File: backend/src/main/scala/microservices/blog/objects/BlogUserInteractionPlannerRequest.scala
// BlogUserInteractionPlannerRequest：博客域博客用户Interaction请求对象。

package com.typesafe.travel.blog.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class BlogUserInteractionPlannerRequest(userId: String, targetUserId: String)
object BlogUserInteractionPlannerRequest:
  given sourceEncoder: Encoder[BlogUserInteractionPlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[BlogUserInteractionPlannerRequest] = deriveDecoder

File: frontend/src/microservices/blog/objects/BlogUserInteractionPlannerRequest.ts
// BlogUserInteractionPlannerRequest：博客域博客用户Interaction请求对象。

export type BlogUserInteractionPlannerRequest = {
  userId: string
  targetUserId: string
}

export const blogUserInteractionPlannerRequestFromJson = (json: string): BlogUserInteractionPlannerRequest =>
  JSON.parse(json) as BlogUserInteractionPlannerRequest

export const blogUserInteractionPlannerRequestToJson = (value: BlogUserInteractionPlannerRequest): string =>
  JSON.stringify(value)